<launch>
  <!-- Chạy move_base với các thông số cấu hình -->
  <node pkg="move_base" type="move_base" name="move_base" output="screen">
  
    <!-- === TẢI CÁC THAM SỐ CẤU HÌNH === -->
     <!-- === TẢI CÁC THAM SỐ CẤU HÌNH === -->
    <rosparam file="$(find robot10)/config/costmap_common_params.yaml" command="load" />
    <rosparam file="$(find robot10)/config/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find robot10)/config/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find robot10)/config/base_local_planner.yaml" command="load" />

    <!-- === THAM SỐ CỦA move_base === -->
    <param name="base_global_planner" value="navfn/NavfnROS" />
    <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS" />
    
       <!-- Tần số lập lại kế hoạch và gửi lệnh điều khiển -->
   <!-- <param name="planner_frequency" value="2.0" /> -->      
   <!-- <param name="controller_frequency" value="10.0" /> -->    
    

   <param name="recovery_behavior_enabled" value="true"/>
    <param name="clearing_rotation_allowed" value="true"/>
    <param name="controller_frequency" value="10.0" />
    <param name="planner_patience" value="5.0" />
    <param name="controller_patience" value="15.0" />
      
      
      <!-- Tùy chọn: khai báo recovery behaviors cụ thể -->
    <rosparam param="recovery_behaviors">
      [
        {name: 'conservative_reset', type: 'clear_costmap_recovery/ClearCostmapRecovery'},
        {name: 'rotate_recovery', type: 'rotate_recovery/RotateRecovery'}
      ]
    </rosparam>

  </node>
</launch>
